tests/Unit/CMClientSettingsComputerRestart.tests.ps1

```diff
@@ -0,0 +1,367 @@
+[System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '')]
+param ()
+
+$script:dscModuleName = 'ConfigMgrCBDsc'
+$script:dscResourceName = 'DSC_CMClientSettingsComputerRestart'
+
+function Invoke-TestSetup
+{
+    try
+    {
+        Import-Module -Name DscResource.Test -Force -ErrorAction 'Stop'
+    }
+    catch [System.IO.FileNotFoundException]
+    {
+        throw 'DscResource.Test module dependency not found. Please run ".\build.ps1 -Tasks build" first.'
+    }
+
+    $script:testEnvironment = Initialize-TestEnvironment `
+        -DSCModuleName $script:dscModuleName `
+        -DSCResourceName $script:dscResourceName `
+        -ResourceType 'Mof' `
+        -TestType 'Unit'
+
+    # Import Stub function
+    $script:moduleRoot = Split-Path -Parent (Split-Path -Parent $PSScriptRoot)
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath 'Stubs\ConfigMgrCBDscStub.psm1') -Force -WarningAction SilentlyContinue
+}
+
+function Invoke-TestCleanup
+{
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+# Begin Testing
+try
+{
+    InModuleScope $script:dscResourceName {
+
+        Describe 'ConfigMgrCBDsc - DSC_CMClientSettingsComputerRestart\Get-TargetResource' -Tag 'Get' {
+            BeforeAll {
+                $clientReturn = @{
+                    RebootNotificationsDialog                 = $true
+                    RebootLogoffNotificationFinalWindow       = 10
+                    RebootLogoffNotificationCountdownDuration = 30
+                    EnforeReboot                              = $true
+                }
+
+                $clientType = @{
+                    Type = '1'
+                }
+
+                $getInput = @{
+                    SiteCode          = 'Lab'
+                    ClientSettingName = 'ClientTest'
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+            }
+
+            Context 'When retrieving Client Policy Settings for Computer Restart' {
+
+                It 'Should return desired results when client settings exist' {
+                    Mock -CommandName Get-CMClientSetting -MockWith { $clientType }
+                    Mock -CommandName Get-CMClientSetting -MockWith { $clientReturn } -ParameterFilter { $Setting -eq 'ComputerRestart' }
+
+                    $result = Get-TargetResource @getInput
+                    $result                                           | Should -BeOfType System.Collections.HashTable
```

Comment:
Spacing before the pipe char lines 70-78.

Reply:
Done